manila/share/manager.py

```diff
@@ -4018,21 +4021,18 @@ def delete_snapshot(self, context, snapshot_id, force=False,
                     resource_id=snapshot_instance_id,
                     exception=excep)
 
-        self.db.share_snapshot_instance_delete(context, snapshot_instance_id)
-        self._delete_snapshot_quota(context, snapshot_ref)
-
-    def _get_snapshot_instances_with_deletion_deferred(self, ctxt):
-        snap_instances = self.db.share_snapshot_instance_get_all_with_filters(
-            ctxt, {'statuses': constants.STATUS_DEFERRED_DELETING})
+        snapshot_instance = self.db.share_snapshot_instance_get(
+            context, snapshot_ref.instance['id'])
+        if snapshot_instance['status'] == (
+                constants.STATUS_ERROR_DEFERRED_DELETING) and not force:
+            return
```

Comment:
if-block not necessary, the condition is never true

Reply:
we allow to delete share/snapshot in all state with Force=True. So if this is error_deferred_deleting state and admin forced to delete, the condition will be forcefully delete otherwise return (and handle in peridoic task)

Reply:
but you have not implemented sth for deleting a snapshot in 'error_deferred_deleting' state for admin, or? If admin deletes the snapshot again, the quota will be deleted twice

Reply:
ok. added states check in manila/share/api.py that if error_defered_deleting and forcefully deleted, we dont clean/delete quota twice.  Force delete is allowed only for admin., so existing policy check of force-delete should be enough to consider deleting deferred state via API.